### ISSUE TITLE
previous valid 0.8 RAML is invalid now based on my strict validating

### ISSUE THREAD

**Issue by sichvoge** [raml-org/raml-js-parser-2#191]
The new implementation of the RAML 0.8 parser is more strict on validation right now. We need to be careful with that as it has a huge impact on current existing files. Here an example: 

[API_Platform-v2.0.3.zip](https://github.com/raml-org/raml-js-parser-2/files/266441/API_Platform-v2.0.3.zip)

The problems with the example file attached:

* in searchable.yaml: trait `query` defines an empty-valued `example:`
* keys with whitespaces (actually no problem according to the spec, but maybe we should warn here)
* invalid media type (seems that it does not recognize the default media type)
* resource path name gets resolved to the wrong path fragment (might be resolved by #188)
* in collection.yaml: description is empty-valued (was not a problem with 0.8)

There are a couple of other problems that makes the project invalid, but these are completely valid cases.

**Comment by petrochenko-pavel-a** [comment]
>invalid media type (seems that it does not recognize the default media type)
it is actually a correct issues which are raised after expansion, (default media type can only be applied only if there is no media types which are already defined in this context)

>keys with whitespaces (actually no problem according to the spec, but maybe we should warn here)
changed to warnings

**Comment by sichvoge** [comment]
For the first related to media types. Please have a look at the example, the errors are on those who don't have a media type. So the default media type should be applied right? For example:

```yaml
      /{apiId}:
        description: An API.
        type: collectionItem
        ...
        get:
          description: Returns an API and all its versions.
          responses:
            200:
              body:
                example: !include examples/api.json
```

The error "invalid media type" is applied to the example node.

**Comment by petrochenko-pavel-a** [comment]
>type: collectionItem

**Comment by sichvoge** [comment]
> It seems that I miss something and I already have fix for #188 but where?

Thats fixed, but not visible yet in the workbench nor the designer. Hence my comment inside the brackets. It was only to have them noted down here. So only FYI

**Comment by sichvoge** [comment]
Another one is #194
